pennylane/devices/qubit/preprocess.py

```diff
@@ -87,13 +96,38 @@ def expand_fn(circuit, max_expansion=10):
     if any(isinstance(o, MidMeasureMP) for o in circuit.operations):
         circuit = qml.defer_measurements(circuit)
 
-    if not all(_stopping_condition(op) for op in circuit.operations):
-        circuit = circuit.expand(depth=max_expansion, stop_at=_stopping_condition)
+    if len(circuit._prep) > 1:
+        raise DeviceError("DefaultQubit2 accepts at most one state prep operation.")
```

Comment:
Is this a constraint we can loosen in the future with a bit of work, or a fairly fundamental restriction of the new device API?

Reply:
Yes we could loosen it in the future.  Instead of raising an error, we could combine multiple state prep operations into a single one.

Reply:
created a [story](https://app.shortcut.com/xanaduai/story/34564/combine-multiple-state-prep-operations-into-a-single-one) in case we want to tackle that in the future.